### PR TITLE
kubectl-all: init

### PR DESCRIPTION
**Pull request by Benjile**
Adds several kubernetes plugins

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -85,6 +85,9 @@ rec {
   fly = pkgs.callPackage ./pkgs/fly.nix
     { source = sources.concourse; };
 
+  kubectl-with-plugins = pkgs.callPackage ./pkgs/kubectl-with-plugins
+    { sources = sources; pkgs = pkgs;};
```
Comment (eonpatapon) [inline default.nix:89]:
```suggestion
    { sources = sources; };
```

File: pkgs/kubectl-with-plugins/default.nix [hunk 2]
```diff
@@ -0,0 +1,70 @@
+{ stdenv
+, fetchzip
+, sources
+, pkgs
+, makeWrapper
+}:
+with pkgs;
+rec {
+  plugins = {
```
Comment (eonpatapon) [inline pkgs/kubectl-with-plugins/default.nix:9]: I would prefer to expose this in a `kubectl-plugins` attribute at the root of the toolbox

File: pkgs/kubectl-with-plugins/default.nix [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        repo = sources.kubectl-topology.repo;
+        rev = "v${version}";
+        sha256 = sources.kubectl-topology.sha256;
+        };
+      vendorSha256 = "18j6lv4aar1fwr9cb186j0mb0kirvhx8m2k98fjqpx50dmiqb695";
+      subPackages = ["cmd/kubectl-topology"];
+      };
+    };
+
+  packages=(builtins.map(x: (builtins.getAttr x plugins)) (builtins.attrNames plugins));
```
Comment (eonpatapon) [inline pkgs/kubectl-with-plugins/default.nix:63]:
```suggestion
  packages = builtins.attrValues plugins;
```

File: pkgs/kubectl-with-plugins/default.nix (hunk 2 above)
Comment (eonpatapon) [reply pkgs/kubectl-with-plugins/default.nix:9]: default.nix
```
{
   ...

   kubectl-plugins = import ./pkgs/kubectl-plugins { ... };    # contains default.nix (attrset of plugins), script.nix

   kubectl-with-plugins = import ./pkgs/kubectl-with-plugins.nix 
     { plugins = kubectl-plugins, ... }; 
}
```

File: pkgs/kubectl-plugins/default.nix [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    postInstall = ''
+      ln -s $out/bin/kail $out/bin/kubectl-tail
+    ''
+    ;
+    src = fetchFromGitHub {
+      owner = sources.kail.owner;
+      repo = sources.kail.repo;
+      rev = "v${version}";
+      sha256 = sources.kail.sha256;
+      };
```
[the remark this answers is not in the document]
Comment (Benjile) [reply pkgs/kubectl-plugins/default.nix]: fixed

File: pkgs/kubectl-plugins/default.nix [hunk 5]
```diff
@@ -0,0 +1,59 @@
+{ sources
+}:
+let 
+  pkgs = import sources.nixpkgs{};
+in
+with pkgs;
+rec {
+  kubectl-node-shell = callPackage ./scripts.nix {source = sources.kubectl-node-shell; cmdName = "kubectl-node_shell";};
+  kubectl-spy        = callPackage ./scripts.nix {source = sources.kubespy; cmdName = "kubespy"; outCmdName = "kubectl-spy";};
```
[the remark this answers is not in the document]
Comment (Benjile) [reply pkgs/kubectl-plugins/default.nix]: fixed

File: pkgs/kubectl-plugins/default.nix [hunk 6]
```diff
@@ -0,0 +1,59 @@
+{ sources
+}:
+let 
+  pkgs = import sources.nixpkgs{};
+in
+with pkgs;
+rec {
+  kubectl-node-shell = callPackage ./scripts.nix {source = sources.kubectl-node-shell; cmdName = "kubectl-node_shell";};
```
[the remark this answers is not in the document]
Comment (Benjile) [reply pkgs/kubectl-plugins/default.nix]: fixed also

**Pull request merged**